linker/shared/src/main/scala/org/scalajs/linker/OutputMode.scala

```diff
@@ -0,0 +1,99 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.linker
+
+/** JavaScript output mode configuration.
+ *
+ *  The output mode specifies what features of modern versions of JavaScript
+ *  are used by the Scala.js linker.
+ *
+ *  - Options whose name is of the form `useX` *force* the linker to use the
+ *    corresponding features, guaranteeing that the specific semantics that
+ *    they provide will be used.
+ *  - Options whose name is of the form `allowX` *allow* the linker to use the
+ *    corresponding features if it supports them. Support for such options can
+ *    be dropped in any subsequent version of the linker, including patch
+ *    versions.
+ */
+final class OutputMode private (
```

Comment:
Should we still call this `OutputMode`? Maybe `OutputConfig` or `OutputOptions` would be more appropriate... Not sure though :S

Reply:
Same problem with the 0.6.x branch. If we rename it, we'll need a forward-compatibility alias in the 0.6.x branch.

If we rename it, I would drop the "output" terminology altogether. Maybe something like `ESFeatures`, since it seems it's going in that direction?

Reply:
I like `ESFeatures`, yes.